components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServlet.java

```diff
@@ -704,29 +704,33 @@ private void handleMissingSessionStateGracefully(HttpServletRequest request, Htt
             response.sendRedirect(getRedirectURL(redirectURL, request));
             return;
         }
+        String clientId = null;
+        try {
+            clientId = extractClientFromIdToken(idTokenHint);
+        } catch (ParseException e) {
+            String msg = "Error occurred while getting application information.";
+            log.error(msg, e);
+            redirectURL = OIDCSessionManagementUtil.getErrorPageURL(OAuth2ErrorCodes.ACCESS_DENIED, msg);
+            response.sendRedirect(getRedirectURL(redirectURL, request));
+        }
         if (!validateIdToken(idTokenHint)) {
             String msg = "ID token signature validation failed.";
-            log.error(msg);
+            log.error(msg + " Client id from id token:" + clientId);
```

Comment:
This is a client error. Should we log here?

Reply:
changed to debug log